### PR TITLE
ENLSVG: Makes sure returns at least two points

### LINKED ISSUE #2610: Make sure the enlsvg path planner always returns at least two points
### Description of the task
as titled, I observed cases of this not happening

### Acceptance criteria

- [ ]

### PR DESCRIPTION
**Pull request by JoshuaWu7**
### Description

Identifies and fixes issue regarding enlsvg path planner returning single point in path.

Observation: We overlooked four edge cases when the start and destination points are too close to each other. The proposed solution is to return a path with the `start` point and `new_end` point. The idea is to have the robot stay relatively stationary and be outside of blocked regions.
image

### Testing Done

Ran all tests to identify failing plays. Recreated tests with motion constraints, start, and destination points. Made sure all tests passed after making the fix.
### Resolved Issues

resolves #2610 
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/ai/navigator/path_planner/enlsvg_path_planner.cpp [hunk 1]
```diff
@@ -90,6 +90,13 @@ std::optional<Path> EnlsvgPathPlanner::findPath(const Point &start,
         return Path({start, end});
     }
 
+    // If the new unblocked points are equal, set the unblocked end point as the end point
+    // of the path
+    if (new_start == new_end)
+    {
+        return Path({start, convertEnlsvgPointToPoint(new_end.value())});
```
Comment (akhilveeraghanta) [inline src/software/ai/navigator/path_planner/enlsvg_path_planner.cpp:97]: should this be new_start?

**Review by akhilveeraghanta — COMMENTED** [review]
lgtm! the oscillations seem to have completely gone, nice work :fire:

File: src/software/ai/navigator/path_planner/enlsvg_path_planner.cpp (hunk 1 above)
Comment (itsarune) [reply src/software/ai/navigator/path_planner/enlsvg_path_planner.cpp:97]: new_start is just for the enlsvg path planner to navigate outside the obstacle (if it starts in one). We want the starting position to be the robot's starting position (aka start)

**Review by itsarune — DISMISSED** [review]
great work!

File: src/software/ai/hl/stp/play/play.cpp [hunk 2]
```diff
@@ -235,9 +235,7 @@ std::unique_ptr<TbotsProto::PrimitiveSet> Play::getPrimitivesFromTactic(
             *(motion_control.mutable_requested_destination()) =
                 *createPointProto(destination);
 
-            // TODO (#2610): we shouldn't have to check that the path has at least two
-            // knots, since the EnlsvgPathPlanner should always find paths with two points
-            if (path.has_value() && path.value().getKnots().size() >= 2)
+            if (path.has_value())
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/play.cpp:238]: we should be able to add a `CHECK (path.value().getKnots().size()>=2)` so make sure we don't have one point paths

**Review by jonathanlew — COMMENTED** [review]
nice work, we don't need that many new tests, though, just 2-3 would be fine

File: src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp [hunk 3]
```diff
@@ -818,3 +818,86 @@ TEST_F(TestEnlsvgPathPlanner, test_enlsvg_path_planner_close_start_end_but_block
...
         {path_points.begin() + 1, path_points.end()}, obstacles);
 }
+
+TEST_F(TestEnlsvgPathPlanner, test_enlsvg_path_planner_simulated_hrvo)
+{
+    Field field        = Field::createSSLDivisionBField();
+    Team friendly_team = Team(Duration::fromMilliseconds(1000));
+    Team enemy_team    = Team(Duration::fromMilliseconds(1000));
+    Ball ball          = Ball(Point(), Vector(), Timestamp::fromSeconds(0));
+    World world        = World(field, ball, friendly_team, enemy_team);
```
Comment (jonathanlew) [inline src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp]: nit: we don't need these variables, all we need is field (replace world.field() with field), same for the tests below

**Review by jonathanlew — COMMENTED** [review]
one more comment

File: src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp (hunk 3 above)
Comment (JoshuaWu7) [reply src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp]: done

**Review by akhilveeraghanta — APPROVED** [review]

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Pull request merged**